server/aesdsocket.c

```diff
@@ -166,10 +176,229 @@ void signal_handler(int signo)
+        {
+            syslog(LOG_ERR, "Error opening %s file: %s", FILENAME, strerror(errno));
+            status = FAILURE;
+            goto exit;
+        }       
+        if (SUCCESS != pthread_mutex_lock(node->thread_mutex))
+        {
+            syslog(LOG_PERROR, "pthread_mutex_lock: %s", strerror(errno));
+            status = FAILURE;
+            goto exit;
+        }
+        /* write the timestamp to the file */
+        written_bytes = write(file_fd, output, strlen(output));
+        if (written_bytes != strlen(output))
+        {
+            syslog(LOG_ERR, "Error writing %s to %s file: %s", output, FILENAME,
+                       strerror(errno));
+            status = FAILURE;
+            pthread_mutex_unlock(node->thread_mutex);
+            goto exit;
+        }
+        if (SUCCESS != pthread_mutex_unlock(node->thread_mutex))
+        {
+            syslog(LOG_PERROR, "pthread_mutex_unlock: %s", strerror(errno));
+            status = FAILURE;
+            goto exit;
+        }
+        status = SUCCESS;
+        close(file_fd);
+    }
+exit:
+     (status == FAILURE) ? (node->thread_complete_success = false) : 
+                           (node->thread_complete_success = true);
+     return thread_node;
+}
+
+/**
+ * @brief Handles socket recv and send data.
+ *
+ * @param thread_node contains thread data.
+ *
+ * @return void *
+ */
+void *recv_and_send_thread(void *thread_node)
+{
+    int recv_bytes = 0;
+    char buffer[MAX_BUFF_LEN] = {'\0'};
+    bool packet_complete = false;
+    int written_bytes = 0;
+    socket_node_t *node = NULL;
+    int status = FAILURE;
+    int file_fd = -1;
+    if (NULL == thread_node)
+    {
+        return NULL;
+    }
+    else
+    {
+        node = (socket_node_t *)thread_node;
+        /* open file in readwrite mode */
+        file_fd = open(FILENAME, O_CREAT|O_RDWR|O_APPEND, 
+                       S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+        if (FAILURE == file_fd)
+        {
+            syslog(LOG_ERR, "Error opening %s file: %s", FILENAME, strerror(errno));
+            status = FAILURE;
+            goto exit;
+        }
+
+        /* loop to receive data until new line is found */
+        do
+        {
+            memset(buffer, 0, MAX_BUFF_LEN);
+            /* recv data from client */
+            recv_bytes = recv(node->connection_fd, buffer, MAX_BUFF_LEN, 0);
+            if (FAILURE == recv_bytes)
+            {
+                syslog(LOG_PERROR, "recv: %s", strerror(errno));
+                status = FAILURE;
+                goto exit;
+            }
+            
+            if (SUCCESS != pthread_mutex_lock(node->thread_mutex))
+            {
+                syslog(LOG_PERROR, "pthread_mutex_lock: %s", strerror(errno));
+                status = FAILURE;
+                goto exit;
+            }
+            /* write the string received to the file */
+            written_bytes = write(file_fd, buffer, recv_bytes);
```

Comment:
@chchbuff what assumption are you making about total packet length vs MAX_BUFF_LEN here?  If a packet was longer than MAX_BUFF_LEN would it be possible for timestamp or other packet prints to be interleaved?

Reply:
My MAX_BUFF_LEN is 1024 and I don't have any assumption on total packet length. As timer is configured for 10secs, I don't think it will be an issue in this assignment if all packet data is received within 10secs. But otherwise yes.